pending/service.go

```diff
@@ -39,6 +42,44 @@ func DefaultService(docSrv documents.Service, repo Repository) Service {
 	}
 }
 
+// Get returns the document associated with docID
+// If status is pending, we return the pending document from pending repo.
+// else, we defer Get to document service.
+func (s service) Get(ctx context.Context, docID []byte, status documents.Status) (documents.Model, error) {
+	if status != documents.Pending {
+		return s.docSrv.GetCurrentVersion(ctx, docID)
+	}
+
+	did, err := contextutil.AccountDID(ctx)
+	if err != nil {
+		return nil, contextutil.ErrDIDMissingFromContext
+	}
+
+	return s.pendingRepo.Get(did[:], docID)
+}
+
+// GetVersion return the specific version of the document
+// We try to fetch the version from the document service, if found return
+// else look in pending repo for specific version.
+func (s service) GetVersion(ctx context.Context, docID, versionID []byte) (documents.Model, error) {
```

Comment:
Wanna add the `GetVersion` to the Service interface above?

Reply:
Oops!
Forgot :-D